src/Packing.cpp

```diff
@@ -0,0 +1,67 @@
+//
+// Created by rolf on 03-03-21.
+//
+
+#include "Packing.h"
+#include "utilities.h"
+#include <cmath>
+
+RobotCommandPayload createLowLevelCommand(const proto::RobotCommand& proto,  const proto::World& world, bool isYellow){
+  float rho = sqrtf(proto.vel().x() * proto.vel().x() + proto.vel().y() * proto.vel().y());
+  float theta = atan2f(proto.vel().y(), proto.vel().x());
+  auto bot = rtt::robothub::utils::getWorldBot(proto.id(),isYellow,world);
+
+  RobotCommand command;
+  command.header = PACKET_TYPE_ROBOT_COMMAND;
+  command.id = proto.id();
+
+  command.doKick = proto.kicker();
+  command.doChip = proto.chipper();
+  command.doForce = proto.chip_kick_forced();
+
+  command.rho = rho;
+  command.theta = theta;
+  command.angle = proto.w();
+
+  if(bot){
+    command.useCameraAngle = true;
+    command.cameraAngle = bot->w();
+  }else{
+    command.useCameraAngle = false;
+    command.cameraAngle = 0.0;
+  }
+  command.kickChipPower = proto.chip_kick_vel();
```

Comment:
Not sure in what range `proto.chip_kick_vel()` lies. 3 bits are reserved for `kickChipPower`.

Reply:
0-6.5 (m/s) in float. I think 3 is really on the low side however; in the past we used to have 8. Was there any reason this was decreased?

Reply:
I think this still needs to be fixed before the robots can kick.

Reply:
There is no specific reason for 3 bits. It just fitted nicely in the packet definition. I think we'll need to wait for the power and resolution of the new kicker before we can decide on the number of bits